addons/website/models/website.py

```diff
@@ -1811,20 +1818,28 @@ def _trigram_enumerate_words(self, search_details, search, limit):
                         search=unaccent(sql.Placeholder('search')),
                         field=unaccent(sql.SQL('value'))
                     )
+                    where_clause_domain = """WHERE lang = {lang}
+                                           AND name = ANY({names})
+                                           AND type = 'model'
+                                           AND value IS NOT NULL
+                                           """
+                    # This method uses hardcoded sql query, it needs filter regarding performance.
+                    # In this case for the is_published field as customer don't have access to
+                    # unpublished objects.
+                    if 'is_published' in model._fields and not self.env.user.has_group('base.group_user'):
```

Comment:
Don't duplicate the condition, add it to a `has_is_published_field` or something. (so you don't need to fix it multiple time, here also need to be `_Model_fields`, not `_fields`